lib/api/src/modules/refs.ts

```diff
@@ -113,46 +123,73 @@ export const init: ModuleFn = ({ store, provider, fullAPI }) => {
     checkRef: async (ref) => {
       const { id, url } = ref;
 
-      const handler = async (response: Response) => {
-        if (response) {
-          const { ok } = response;
-
-          if (ok) {
-            return response.json().catch((error: Error) => ({ error }));
-          }
-        } else {
-          logger.warn('an auto-injected ref threw a cors-error');
-        }
+      const loadedData: { error?: Error; stories?: StoriesRaw } = {};
 
-        return false;
-      };
-
-      const [stories, metadata] = await Promise.all([
+      const [included, omitted, iframe] = await allSettled([
         fetch(`${url}/stories.json`, {
           headers: {
             Accept: 'application/json',
           },
           credentials: 'include',
-        })
-          .catch(() => false)
-          .then(handler),
-        fetch(`${url}/metadata.json`, {
+        }),
+        fetch(`${url}/stories.json`, {
           headers: {
             Accept: 'application/json',
           },
-          credentials: 'include',
-          cache: 'no-cache',
-        })
-          .catch(() => false)
-          .then(handler),
+          credentials: 'omit',
+        }),
+        fetch(`${url}/iframe.html`, {
+          credentials: 'omit',
+        }),
       ]);
 
+      const handle = async (request: Promise<Response> | false) => {
+        if (request) {
+          return Promise.resolve(request)
+            .then((response) => (response.ok ? response.json() : {}))
+            .catch((error) => ({ error }));
+        }
+        return {};
+      };
+
+      if (!included && !omitted && !iframe) {
+        loadedData.error = {
+          message: dedent`
+            Error: Loading of ref failed
+              at fetch (lib/api/src/modules/refs.ts)
+
+            URL: ${url}
+
+            We weren't able to load the above URL,
+            it's possible a CORS error happened.
+
+            Please check your dev-tools network tab.
+          `,
+        } as Error;
+      } else {
+        const credentials = !omitted ? 'include' : 'omit';
+
+        const [stories, metadata] = await Promise.all([
+          handle(omitted || included),
+          handle(
+            fetch(`${url}/metadata.json`, {
+              headers: {
+                Accept: 'application/json',
+              },
+              credentials,
+              cache: 'no-cache',
+            })
+          ),
+        ]);
```

Comment:
If both `!included` and `!omitted` (i.e. we weren't able to get `stories.json` at all), do we bother with `metadata.json`?

Reply:
good point!